src/test/java/edu/kis/powp/jobs2d/command/ImmutableCompoundCommandTest.java

```diff
@@ -0,0 +1,40 @@
+package edu.kis.powp.jobs2d.command;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ImmutableCompoundCommandTest {
+
+    @Test
+    void iterator() {
+
+        List<DriverCommand> dcList = new ArrayList<>();
+        SetPositionCommand spc = new SetPositionCommand(10, 10);
+        dcList.add(spc);
+        dcList.add(new OperateToCommand(10, 10));
+        dcList.add(new SetPositionCommand(10, 10));
+        dcList.add(new OperateToCommand(10, 10));
```

Comment:
To test "deep immutability" create and add a mutable simple ComplexComand inside this CompoundCommand and then change it.

Reply:
Test updated